### ISSUE TITLE
Range filter not working when writting query with ejs aggregations

### ISSUE THREAD

**Issue by pawansmartdata** [elastic/elasticsearch#18152]
My requirement is to add a filter in to the query by using ejs but i couldn't get success on that.
I have created a Json query for the same that is working for me but i need to convert that Json in form of ejs aggregation.
Below is my JSON query:-
 
{
    "aggs": {
        "parentsku": {
            "terms": {
                "field": "parentSku",
                "order": {
                    "_term": "asc"
                },
                "size": 2
            },
            "aggs": {
                "cost": {
                    "stats": {
                        "field": "directCost"
                    }
                },
                "sales": {
                    "stats": {
                        "field": "revenu"
                    }
                },
                "quantity": {
                    "stats": {
                        "field": "quantity"
                    }
                }
            }
            
        }
    },
     "query": {
        "filtered": {
            "filter": {
                "range": {
                    "quantity": {
                        "gte": "500"
                    }
                }
            }
        }
    }
}

This query works for me when i check it with Postman client of google crome but i couldn't convert this is the form ejs aggregation and I have to use all the queries in form of ejs aggregation only.
Below is my ejs aggregation query:-

var myquery= ejs.Request()
                                 .agg(ejs.TermsAggregation('parentsku').field(skufield.value 					|| 'parentSku').order("_term", "asc").size(50)
                                  .agg(ejs.StatsAggregation('cost').field('directCost'))
					.agg(ejs.StatsAggregation('sales').field('revenu'))
                                  .agg(ejs.StatsAggregation('quantity').field('quantity'))
                                );

I only need to add below part of query with above aggregation query and it will work for me.

 "query": {
        "filtered": {
            "filter": {
                "range": {
                    "quantity": {
                        "gte": "500"
                    }
                }
            }
        }
    }

In short i want to convert above Json query in form of aggregration.I tried many alternates for that below is one from them 

var query= ejs.Request().query(ejs.FilterAggregation('filtered').filter(ejs.RangeFilter('quantity').gt(200)))
but it is not working exception is like "Argument must be query" it means "ejs.FilterAggregation('filtered').filter(ejs.RangeFilter('quantity').gt(200))" is not a query.

**Comment by jasontedor** [comment]
Elastic reserves GitHub for bug reports and feature requests but provides a [Discourse forum](https://discuss.elastic.co) for general questions.

**Comment by pawansmartdata** [comment]
I have asked same question there too but no one replied there..
